### PR TITLE
Move to Go 1.13

### PR DESCRIPTION
**Pull request by rminnich**
The Unwrap() is compelling and is used in another PR.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -126,9 +134,36 @@
   pruneopts = "NUT"
   revision = "fc88cf888a3fa99ecc23d1efc1a44284268457d3"
 
+[[projects]]
+  digest = "1:11c6c696067d3127ecf332b10f89394d386d9083f82baf71f40f2da31841a009"
+  name = "github.com/hashicorp/hcl"
```
Comment (hugelgupf) [inline Gopkg.lock:139]: ron what is this stuff?

you have *something* in your u-root directory that _always_ screws up your dep update. remove that something.

**Comment by hugelgupf** [comment]
u-root.go also compares version numbers. Remove go1.12 from the recommended list as well?

**Comment by rjoleary** [comment]
Once you submit this, I'll tag it with the magic label to rebuild the Docker image. Only then will it actually be using go1.13 in the CI.

**Comment by rminnich** [comment]
I thought I had removed it ...

**Comment by rminnich** [comment]
yeah it's gone. 

dep drives me nuts. I don't have any .go files for it to get confused by. Oh well, try again.

**Comment by rjoleary** [comment]
```
--- FAIL: Test_Compress (0.00s)
    --- FAIL: Test_Compress/Basic_Compress (0.00s)
        gzip_test.go:69: Compress() = "\x1f\x8b\b\x00\x00\tn\x88\x02\xff\nI-.Q\x80\x13\x00\x00\x00\x00\xff\xff\x03\x00G?\xfc\xcc\x0e\x00\x00\x00", want "\x1f\x8b\b\x00\x00\tn\x88\x02\xff\nI-.Q\x80\x13\x00\x00\x00\x00\xff\xff\x01\x00\x00\xff\xffG?\xfc\xcc\x0e\x00\x00\x00"
    --- FAIL: Test_Compress/Zeros (0.00s)
        gzip_test.go:69: Compress() = "\x1f\x8b\b\x00\x00\tn\x88\x02\xff2 \x1d\x00\x00\x00\x00\xff\xff\x03\x00R6\xe3\xeb3\x00\x00\x00", want "\x1f\x8b\b\x00\x00\tn\x88\x02\xff2 \x1d\x00\x00\x00\x00\xff\xff\x01\x00\x00\xff\xffR6\xe3\xeb3\x00\x00\x00"
    --- FAIL: Test_Compress/Nil (0.00s)
        gzip_test.go:69: Compress() = "\x1f\x8b\b\x00\x00\tn\x88\x04\xff\x00\x00\x00\xff\xff\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00", want "\x1f\x8b\b\x00\x00\tn\x88\x04\xff\x00\x00\x00\xff\xff\x01\x00\x00\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00"
FAIL
FAIL	github.com/u-root/u-root/pkg/gzip	0.215s
```

Perhaps the compression algorithm changed? The test shouldn't assume the output is an exact match and instead perform a compress+decompress and compare to the original uncompressed data.

File: RELEASES [hunk 2]
```diff
@@ -20,7 +20,7 @@ A new release happens every 6 weeks:
 - checklicenses command is generalized to be used by other projects.
 - Tested with:
   - Linux 4.17
-  - Golang 1.12
+  - Golang 1.13
```
Comment (hugelgupf) [inline RELEASES:23]: I don't think you want to change this one

**Comment by rminnich** [comment]
our tests are not handling this one too well ... I'll try to look later today.

**Comment by hugelgupf** [comment]
#1659 has a working dep update.

**Comment by hugelgupf** [comment]
Pushed an update now that the dep update has been merged. Building the new docker images now.

**Comment by hugelgupf** [comment]
Alright, something's still broken. Gonna put this aside for now, I have to work on mutiboot.

**Comment by hugelgupf** [comment]
Just gonna highlight that last commit here for a sec:

> The source-built initramfs contains a 386-only toolchain. Up until Go
1.12, apparently it was fine to run this test, because the 386 Go
toolchain seemed to be compatible with amd64 systems.
>
> However, in Go 1.13 there's some 386-only stuff that is incompatible
with amd64 systems, which results in a "fatal error: stack growth after
fork" when trying to run the 386 toolchain on amd64.

**Pull request merged**